packages/data-union-solidity/Makefile

```diff
@@ -24,7 +24,7 @@ $(JQ): $(BUILD_TOOL_DIR)
 
 # find solc downloads at https://github.com/ethereum/solc-bin/
 # line is like "pragma solidity 0.8.6;"
-SOLC_VERSION = $(shell grep pragma contracts/DataUnionMainnet.sol |cut -c 34-38)
+SOLC_VERSION = $(shell grep --color=never "pragma solidity" contracts/DataUnionMainnet.sol |sed 's/[^0-9]*\([0-9.]*\).*/\1/')
```

Comment:
grep `--color` flag is unnecessary.

Grep expression could check line start `^pragma solidity` and end `;$`

Quantifier in sed expression should be `+`.

Is the purpose of `^` in `[^0-9]` to match the start of the line in sed expression?

Reply:
my grep colors things by default, probably some setting that might exist in other systems too.

Adding ^ and $ are good ideas.

I guess sed expression could as well match the whole line, since it's completely known.

Reply:
+ doesn't seem to work in macosx

Reply:
`+` doesn't seem to work in macosx

Reply:
also there's no `grep -Po` which would enable picking the thing with `\K`, alas macosx grep doesn't support it.